test/all.spec.ts

```diff
@@ -318,6 +334,95 @@ describe('validation', () => {
 		};
 		expect(f).to.throw();
 	});
+
+	it('should not fail when build config specifies valid network', async () => {
+		const f = () => {
+			compose.normalize({
+				version: '2.4',
+				networks: {
+					mynet: {},
+				},
+				services: {
+					main: {
+						build: { context: '.', network: 'mynet' },
+					},
+				},
+			});
+		};
+		expect(f).to.not.throw();
+	});
+
+	it('should fail when build config specifies invalid network', async () => {
+		const f = () => {
+			compose.normalize({
+				version: '2.4',
+				networks: {
+					othernet: {},
+				},
+				services: {
+					main: {
+						build: { context: '.', network: 'mynet' },
+					},
+				},
+			});
+		};
+		expect(f).to.throw();
+	});
+
+	it('should support extension fields', async () => {
+		const f = () => {
+			const c = compose.normalize({
+				version: '2.4',
+				services: {
+					main: {
+						build: '.',
+						'x-extension': true,
+					},
+				},
+			});
+		};
+		expect(f).to.not.throw();
+	});
+
+	it('should throw when long syntax tmpfs mounts are specified', async () => {
+		const f = () => {
+			const c = compose.normalize({
+				version: '2.4',
+				services: {
+					main: {
+						build: '.',
+						volumes: [{ type: 'tmpfs', target: '/tmp2' }],
```

Comment:
What is the expected error message in this case? Does it point the user to the correct way to do this?

Reply:
currently:

`Tmpfs mounts are not allowed`

should we point them to the tmpfs field instead?

Reply:
Well, the error is misleading because we actually support tmpfs mounts, just with different notation, so yes, at the very least we should point the user to the correct way to do this.

However, given we do support tmpfs mounts and user's intent is obvious, is there a reason not to automatically translate internally to the accepted notation instead of rejecting their input?

Reply:
true, we could do automatic translation, and then similar to the volume options we just have to error on the tmpfs options as well

Reply:
implemented in [`5dbedda` (#59)](https://github.com/balena-io-modules/resin-compose-parse/pull/59/commits/5dbeddaa9ff85ce3d88eac4771d3d64aa55edc7b)